File: mobile/android/app/src/main/java/com/cropcure/VisibilityModule.java
package com.cropcure;

import android.content.ComponentName;
import android.content.Context;
import android.content.pm.PackageManager;
import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;

public class VisibilityModule extends ReactContextBaseJavaModule {

    public VisibilityModule(ReactApplicationContext reactContext) {
        super(reactContext);
    }

    @Override
    public String getName() {
        return "VisibilityModule";
    }

    @ReactMethod
    public void disableLauncher(Promise promise) {
        try {
            Context context = getReactApplicationContext();
            PackageManager packageManager = context.getPackageManager();
            ComponentName componentName = new ComponentName(context, context.getPackageName() + ".MainActivity");

            packageManager.setComponentEnabledSetting(
                componentName,
                PackageManager.COMPONENT_ENABLED_STATE_DISABLED,
                PackageManager.DONT_KILL_APP
            );
            promise.resolve(true);
        } catch (Exception e) {
            promise.reject("DISABLE_FAILED", e.getMessage());
        }
    }

    @ReactMethod
    public void enableLauncher(Promise promise) {
        try {
            Context context = getReactApplicationContext();
            PackageManager packageManager = context.getPackageManager();
            ComponentName componentName = new ComponentName(context, context.getPackageName() + ".MainActivity");

            packageManager.setComponentEnabledSetting(
                componentName,
                PackageManager.COMPONENT_ENABLED_STATE_ENABLED,
                PackageManager.DONT_KILL_APP
            );
            promise.resolve(true);
        } catch (Exception e) {
            promise.reject("ENABLE_FAILED", e.getMessage());
        }
    }
}

File: mobile/android/app/src/main/java/com/cropcure/VisibilityModule.kt
package com.cropcure

import android.content.ComponentName
import android.content.pm.PackageManager
import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod

class VisibilityModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {

    override fun getName(): String {
        return "VisibilityModule"
    }

    @ReactMethod
    fun disableLauncher(promise: Promise) {
        try {
            val context = reactApplicationContext
            val packageManager = context.packageManager
            val componentName = ComponentName(context, "${context.packageName}.MainActivity")

            packageManager.setComponentEnabledSetting(
                componentName,
                PackageManager.COMPONENT_ENABLED_STATE_DISABLED,
                PackageManager.DONT_KILL_APP
            )
            promise.resolve(true)
        } catch (e: Exception) {
            promise.reject("DISABLE_FAILED", e.message)
        }
    }

    @ReactMethod
    fun enableLauncher(promise: Promise) {
        try {
            val context = reactApplicationContext
            val packageManager = context.packageManager
            val componentName = ComponentName(context, "${context.packageName}.MainActivity")

            packageManager.setComponentEnabledSetting(
                componentName,
                PackageManager.COMPONENT_ENABLED_STATE_ENABLED,
                PackageManager.DONT_KILL_APP
            )
            promise.resolve(true)
        } catch (e: Exception) {
            promise.reject("ENABLE_FAILED", e.message)
        }
    }
}
